Polish Custom HTML block

ZebulanStanphill opened the pull request:
## Description
This PR polishes the Custom HTML block. Notable changes:

- Switch from `PlainText` v1 to using `TextareaAutosize` directly.
- Refresh UI and increase font size to match the latest Edit as HTML style introduced in #25539.
- Move `defaultStyles` into a constant defined outside of the component, so it doesn't get recreated every time the `useSelect` callback runs.
- Rename `wp-block-html` CSS class to `block-library-html`.
- Remove usage of `components-tab-button` CSS class that appears to have no visible effect or purpose.
- Remove unnecessary `margin-bottom` editor style.
- Reorder imports alphabetically.

## Screenshots

### Unselected

#### Before
![image](https://user-images.githubusercontent.com/19592990/94698649-22e0b580-02ff-11eb-873e-6681b0e0ce0a.png)

#### After
![image](https://user-images.githubusercontent.com/19592990/94700002-98995100-0300-11eb-990d-f8bdf2d8f936.png)

### Selected; textarea NOT focused

#### Before
![image](https://user-images.githubusercontent.com/19592990/94700805-8370f200-0301-11eb-92fe-851bca5bad39.png)

#### After
![image](https://user-images.githubusercontent.com/19592990/94700179-cf6f6700-0300-11eb-9643-62822373462b.png)

### Selected; textarea focused

#### Before
![image](https://user-images.githubusercontent.com/19592990/94698538-09d80480-02ff-11eb-92c3-04593cef94a7.png)

#### After
![image](https://user-images.githubusercontent.com/19592990/94700028-a222b900-0300-11eb-9180-0fae63eae951.png)

### Navigation mode AKA select tool

#### Before
![image](https://user-images.githubusercontent.com/19592990/94698602-16f4f380-02ff-11eb-9bee-0d9c75b9f002.png)

#### After
![image](https://user-images.githubusercontent.com/19592990/94700284-e57d2780-0300-11eb-974f-13f50449dd9c.png)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

ZebulanStanphill commented on packages/block-library/src/html/editor.scss:
```diff
@@ -23,8 +14,32 @@
 
-		&:focus {
-			box-shadow: none;
+		&:focus::after {
```
This is copied from [here](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/block-list/style.scss#L72). Any ideas for how to simplify this and avoid unnecessary repetition?

karmatosed commented:
Thanks for iterating here @ZebulanStanphill. My main wonder is about the unselected state. It feels a little strong and I wonder if the border is needed?

ZebulanStanphill commented:
@karmatosed Since the Custom HTML block in non-preview mode isn't representative of the front-end, I figured I should make it stand out using the same borders as the block toolbar and placeholders, to make it clear that it's part of the block UI, not the content.

ZebulanStanphill commented:
It also helps to distinguish the block from the Code block.

karmatosed commented:
Thanks for that clarification, for me @ZebulanStanphill it still gives it an active focus state feel. What do you think of iterating around this?

ZebulanStanphill commented:
@karmatosed I'm fine with using a lighter gray for now (we can always change it later). How's this? It's the same gray used by most text input fields in the inspector.

![image](https://user-images.githubusercontent.com/19592990/80245395-3bea8980-8630-11ea-97e1-004dc5602c2a.png)

karmatosed commented:
@ZebulanStanphill as this isn't an input I would probably lean against styling it the same. Is that aa white background being added or just the theme?

ZebulanStanphill commented:
@karmatosed Um... but it is a text input. Not an HTML `<input>`, but an input nonetheless. And unlike most blocks, where the text inputs double as previews, this text input is just a text input. And yes, the white background is being added by editor styles (both before and after this PR).

Whatever the case, it just occurred to me that this should probably be styled the same way as the "Edit as HTML" view that most blocks have. That editor lacks any border at all, regardless of whether your text cursor is in it or not. That seems like a bug to me, but for now I've made the Custom HTML block's HTML view match.

![image](https://user-images.githubusercontent.com/19592990/80402725-f1614b00-8883-11ea-8b35-8fe83f848e50.png)

Is this what you want? We can change both it and the standard Edit as HTML view simultaneously later on, but now they're at least consistent, which they weren't before.

karmatosed commented:
@ZebulanStanphill for me that makes more sense yes.

ZebulanStanphill commented:
I rebased this PR, and while I was at it, I reverted some class name changes that went against the [coding guidelines](https://developer.wordpress.org/block-editor/contributors/develop/coding-guidelines/#naming).

This PR is ready for review.

karmatosed commented:
@ZebulanStanphill just testing this on Twenty Nineteen I don't see any border, was that your intention as I know you were looking to have a faint one kept. For comparison above is a code block.

![image](https://user-images.githubusercontent.com/253067/82476762-6e1fc780-9ac6-11ea-96b0-603fea59f357.png)

ZebulanStanphill commented:
@karmatosed Actually, I was most recently trying to make it match the style of blocks when you use the Edit as HTML option. That doesn't have any visible borders. (In a future PR I would update both to use a visible border.)

![image](https://user-images.githubusercontent.com/19592990/82489178-d1016680-9aa6-11ea-8343-5971b7773088.png)

Does that sound good to you?

enriquesanchez commented:
>I was most recently trying to make it match the style of blocks when you use the Edit as HTML option.

Could this add some confusion by making it look like the 'Edit as HTML' mode? I feel these two are different things, one is a block, the other is an editing mode and should look different, no?

ZebulanStanphill commented:
@enriquesanchez Functionally, they're exactly the same. In both cases, you're editing the HTML of the block. I see no reason to style them differently.

ZebulanStanphill commented:
Rebased to keep styles working after changes in #23454 and #23648.

ZebulanStanphill commented:
Rebased after the changes in #26055.

ZebulanStanphill committed:
Polish Custom HTML block.

ZebulanStanphill commented:
Several of the changes in this PR have already made it into `trunk` via other PRs, and the remaining changes would be better handled as smaller, more focused PRs, so I'm closing this one.
